src/cpufreq.c

```diff
@@ -53,9 +150,9 @@ static int cpufreq_init (void)
 			(num_cpu == 1) ? "" : "s");
 
 	if (num_cpu == 0)
-		plugin_unregister_read ("cpufreq");
+		return -1;
```

Comment:
1) Please, look on https://github.com/collectd/collectd/pull/1730#discussion_r65807467
2) Because it's more correct way regarding collectd documentation.